src/basilisp/core.lpy

```diff
@@ -6891,12 +6897,16 @@
          (~'_record_lrepr [~this-gs py-kwargs#]
           (let [{print-meta :print_meta} (py->lisp py-kwargs#)
 
-                ns-name   (name *ns*)
-                qual-name (.- ~type-name ~'__qualname__)]
+                ;; Accessing this type name directly as `(.-__qualname__ this)`
+                ;; can fail in cases where the type name is munged and starts
+                ;; with a leading double underscore. Python apparently mangles
+                ;; the name as it would for any object field prefixed with a
+                ;; dunder.
+                qual-name (.- (python/type ~this-gs) ~'__qualname__)]
             (cond->> (->> (mapcat identity ~this-gs)
                           (apply hash-map)
                           (repr)
-                          (str "#" ns-name "." qual-name))
+                          (str "#" ~(name *ns*) "." qual-name))
```

Comment:
This was a bizarre bug where the Lisp representation would always be in terms of the current `*ns*` not the namespace the record was defined in.